linera-core/src/chain_worker/state/mod.rs

```diff
@@ -361,14 +364,7 @@ where
                 missing_blob_ids.push(*blob_id);
             }
         }
-        let blobs = self.storage.read_blobs(&missing_blob_ids).await?;
-        let mut not_found_blob_ids = Vec::new();
-        for (blob, blob_id) in blobs.into_iter().zip(missing_blob_ids) {
-            match blob {
-                None => not_found_blob_ids.push(blob_id),
-                Some(blob) => found_blobs.push(blob),
-            }
-        }
+        let not_found_blob_ids = self.storage.missing_blobs(&missing_blob_ids).await?;
```

Comment:
This change will break `get_blobs`. Some of the blobs in `missing_blob_ids` will be found, some will not. In the current implementation you will ignore and not return any of the blobs that are in storage.